islandora_transforms/BPL_MODS_to_solr.xslt

```diff
@@ -0,0 +1,146 @@
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'all_subject_descendants'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Non-otherTyped relatedItem/titleInfo/title -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:relatedItem[not(@otherType)]/mods:titleInfo/mods:title">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'relatedItem_not_otherType_titleInfo_title'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Non-typed note -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:note[not(@type)]">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'note_no_type'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Both types of call numbers, for advanced search -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:identifier[@type='local-call-number'] | mods:identifier[@type='non-marc-call-number']">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'all_call_numbers'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Both types of filing suffixes, for advanced search -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:identifier[@type='local-filing-suffix'] | mods:identifier[@type='non-marc-filing-suffix']">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'all_filing_suffixes'"/>
+      <xsl:with-param name="content" select="normalize-space()"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Concatenate title/nonSort with title -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:titleInfo[mods:nonSort and mods:title]">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'titleInfo_title_concatenated_nonSort'"/>
+      <xsl:with-param name="content" select="concat(normalize-space(mods:nonSort/text()), ' ', normalize-space(mods:title/text()))"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Concatenate physicalDescription/extent with the extent's @unit -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:physicalDescription/mods:extent[@unit]">
+    <xsl:call-template name="write_bpl_field">
+      <xsl:with-param name="field_name" select="'physicalDescription_extent_concatenated_unit'"/>
+      <xsl:with-param name="content" select="concat(normalize-space(text()), ' ', normalize-space(@unit))"/>
+    </xsl:call-template>
+  </xsl:template>
+
+  <!-- Concatenate geographic elements together using double-dashes -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:subject">
```

Comment:
All the logic in the above templates that focus on mods:subject and it's children should either be nested in this template or there should be calls to `<xsl:apply-templates>` on it's children, potentially with different modes to ensure they all get called as appropriate.